src/components/basic/StaticTable/VerticalTable.tsx

```diff
@@ -18,51 +18,147 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+import { useState } from 'react'
 import { Typography } from '@mui/material'
-import { type TableType } from './types'
+import EditIcon from '@mui/icons-material/Edit'
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline'
+import CloseIcon from '@mui/icons-material/Close'
+import { TableType } from './types'
+import { Input } from '../Input'
+import { Tooltips } from '../ToolTips'
 
-export const VerticalTable = ({ data }: { data: TableType }) => (
-  <table style={{ width: '100%', borderCollapse: 'collapse' }}>
-    <thead>
-      <tr>
-        {data.head.map((col, c) => (
-          <th
-            key={c}
-            style={{
-              backgroundColor: '#ecf0f4',
-              textAlign: 'left',
-              padding: '10px 15px',
+export const VerticalTable = ({
+  data,
+  handleEdit,
+}: {
+  data: TableType
+  handleEdit?: (inputValue: string) => void
+}) => {
+  const [inputField, setInputField] = useState<any>(null)
+  const [inputValue, setInputValue] = useState('')
+  const [inputErrorMsg, setInputErrorMessage] = useState('')
+
+  const handleEditFn = (
+    e: React.SyntheticEvent,
+    row: number,
+    column: number
+  ) => {
+    e.stopPropagation()
+    setInputErrorMessage('')
+    setInputValue(data?.edit?.[row][column].inputValue ?? '')
+    setInputField({ row: row, column: column })
+  }
+
+  const addInputValue = (value: string, row: number, column: number) => {
+    setInputValue(value)
+    const editField = data?.edit?.[row][column]
+    editField?.isValidate && setInputErrorMessage(!editField?.isValidate(value.trim()) ? (editField?.errorMessage ?? '') : '')
```

Comment:
I like the solution. But one tiny change: can you please call this `isValid` instead of `isValidate` -  so the naming is consistent through the library